packages/story-editor/src/app/story/effects/useHistoryEntry.js

```diff
@@ -23,28 +23,83 @@ import { useEffect, useRef } from '@web-stories-wp/react';
  * Internal dependencies
  */
 import { useHistory } from '../../history';
+import deleteNestedKeys from '../utils/deleteNestedKeys';
+
+// @todo REMOVE ROTATION ANGLE! For testing only!
+const ELEMENT_PROPS_TO_IGNORE = ['resource.baseColor', 'rotationAngle'];
 
 // Record any change to core variables in history (history will know if it's a replay)
 function useHistoryEntry({ story, current, pages, selection, capabilities }) {
   const {
+    state: { currentEntry },
     actions: { stateToHistory },
   } = useHistory();
 
+  const currentHistoryEntryRef = useRef();
+  useEffect(() => {
+    if (currentEntry) {
+      currentHistoryEntryRef.current = JSON.parse(JSON.stringify(currentEntry));
+    }
+  }, [currentEntry]);
+
   const currentPageIndexRef = useRef();
   const selectedElementIdsRef = useRef();
   useEffect(() => {
     currentPageIndexRef.current = current;
     selectedElementIdsRef.current = selection;
   }, [current, selection]);
 
-  useEffect(() => {
-    stateToHistory({
-      story,
-      current: currentPageIndexRef.current,
-      selection: selectedElementIdsRef.current,
-      pages,
-      capabilities,
+  const deleteKeysFromPages = (list) => {
+    // Create a copy of the list not to influence the original.
+    return JSON.parse(JSON.stringify(list)).map((page) => {
+      const { elements } = page;
+      return {
+        ...page,
+        elements: elements.map((element) =>
+          deleteNestedKeys(element, ELEMENT_PROPS_TO_IGNORE)
+        ),
+      };
     });
+  };
+
+  useEffect(() => {
+    // There are some element properties that should not influence history.
```

Comment:
@barklund WDYT of this approach? It exposes `currentEntry` to the "world" but at the same time leaves the reducer completely as it is -- since the reducer shouldn't be aware of when it should add a history entry and when not.

This basically allows defining any nested (or not) element properties that should not create a new history entry. Since these propeties should still be saved into history though, the change is only for the case when _only_ the ignored property changes.

Currently, for testing purposes, `rotationAngle` is ignored. 

LMK if you have any thoughts on this approach.